elements/lisk-transactions/src/utils/sign_and_validate.ts

```diff
@@ -36,7 +36,7 @@ export const validateSignature = (
 			? new TransactionError(
 					`Failed to validate signature ${signature}`,
 					id,
-					'.signature',
+					'.signatures',
```

Comment:
its validating signature, I know where is the confusion coming from.. maybe we can remove the whole data path.. as the error is not corresponding to `signatures` array.

Reply:
Hmm I think it's good to indicate to the user what property is the offending one so it can pay attention to it.

Reply:
All utils should not return a `TransactionError` but something else such as `ValidationError` or even a plain `Error`. The framework who then use this package can then catch the error and throw a different one with some context, a `TransactionError` in the case of `BaseTransaction`.
So here it's not really about the message, which is fine, it's about the error handling strategy, which is not fine.
The `TransactionError` thrown in `BaseTransaction` can indeed show the path of the invalid property.
My experience in writing libs and framework shown me a good framework error message should also include any help it can to resolve the problem, such as where the developer can find the right value, or even in case of complicated issue, a link to the documentation where the problem and its solution -if any- is exposed.

Reply:
Also, i think this issue is more general one https://github.com/LiskHQ/lisk-sdk/issues/1439 although the description code itself is bit outdated.
If you have further idea, please add to this issue